core/src/test/java/org/apache/calcite/test/fuzzer/RexProgramFuzzyTest.java

```diff
@@ -334,89 +323,8 @@ private void assertSimplifies(String expected, RexNode node) {
-    assertSimplifies("null", not(nullBool));
-    assertSimplifies("false", not(trueLiteral));
-    assertSimplifies("true", not(falseLiteral));
-    assertSimplifies("false", isFalse(nullInt));
-    // TODO: -42 ?
-    assertSimplifies("-(42)", unaryMinus(literal(42)));
-    assertSimplifies("42", unaryMinus(unaryMinus(literal(42))));
-
-    assertSimplifies("IS NOT FALSE(?0.bool0)", isFalse(isFalse(vBool())));
-    assertSimplifies("IS NOT FALSE(?0.bool0)", isFalse(isFalse(isFalse(isFalse(vBool())))));
-    assertSimplifies("null", unaryMinus(nullInt));
-    assertSimplifies("IS NULL(?0.int0)", isNull(vInt()));
-    assertSimplifies("true", isNotNull(vIntNotNull()));
-    assertSimplifies("false", isFalse(vInt()));
-    assertSimplifies("false", isFalse(vIntNotNull()));
-    assertSimplifies("IS FALSE(?0.bool0)", isFalse(vBool()));
-    assertSimplifies("IS FALSE(?0.bool0)", isFalse(vBool()));
-    assertSimplifies("NOT(?0.notNullBool0)", isFalse(vBoolNotNull()));
-    // {$34=-1, $31=-1, $13=true}, COALESCE(=($34, null), IS NOT FALSE($31), $13) yielded true,
-    // and $31 yielded -1
-    assertSimplifies("true",
-        coalesce(
-            eq(vInt(34), nullInt),
-            isNotFalse(vInt(31)), vBool(13)));
-    assertSimplifies("true", isNotTrue(nullInt));
-    assertSimplifies("true", isNotFalse(nullInt));
-    assertSimplifies("false", and(falseLiteral, falseLiteral));
-    assertSimplifies("false", and(falseLiteral, trueLiteral));
-    assertSimplifies("false", and(falseLiteral, nullInt));
-    assertSimplifies("true", and(trueLiteral, trueLiteral));
-    assertSimplifies("null", and(trueLiteral, nullInt));
-    assertSimplifies("null", and(nullInt, nullInt));
-    assertSimplifies("true",
-        or(
-            ne(nullBool, vBoolNotNull(2)), // => NULL
-            eq(isNotNull(nullBool), falseLiteral), // => TRUE
-            falseLiteral));
-    assertSimplifies("true",
-        le(coalesce(falseLiteral, nullBool),
-            or(
-                ne(nullBool, vBoolNotNull(2)),
-                le(nullBool, nullBool),
-                eq(isNotNull(nullBool), isFalse(nullBool)),
-                vBool(12))));
-    assertSimplifies("OR(null, $11, AND(null, $14))",
-        or(nullInt,
-            or(
-                vBool(11),
-                and(nullInt,
-                    vBool(14)), nullInt)));
-    assertSimplifies("true", gt(trueLiteral, falseLiteral));
-    assertSimplifies("false",
-        and(
-            vBool(13),
-            isNotNull(and(trueLiteral, trueLiteral, nullInt)),
-            isNotNull(coalesce(nullInt)),
-            vBoolNotNull()));
-    assertSimplifies("null", ge(nullInt, falseLiteral));
-    assertSimplifies("false",
-        isNotNull(
-            ge(
-                coalesce(nullInt),
-                falseLiteral)));
-    assertSimplifies("true",
-        isNull(
-            ge(
-                coalesce(nullInt),
-                falseLiteral)));
-  }
```

Comment:
Curious why these were removed as these look useful (assuming they fail).

Reply:
1) They do not belong to fuzzer.
The gems found by fuzzer belong to another class.

2) Those are the cases I have found in my heavily custom RexSimplify, so it does not make sense to keep those items. We can easily find new ones which are related to Calcite codebase.

Reply:
Thanks for clarifying :)